Story/quad 139

bharathsharma opened the pull request:
Workspace count to be added,
Some UI modifications to be done

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/mapper/impl/NetworkMapper.java:
```diff
@@ -53,6 +53,7 @@ public INetwork getNetwork(String networkId) throws QuadrigaStorageException
 			network = networkFactory.createNetworkObject();
 			network.setNetworkId(networksDTO.getNetworkid());
 			network.setNetworkName(networksDTO.getNetworkname());
+		//	network.setCreatedDate(networkDTO.getCreateddate());
```
Why did you comment this out?

bharathsharma replied:
This was a comment added by me and was not originally present as part of code. I have removed it

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
 @Controller
```
please use correct code style guide

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+	private static final Logger logger = LoggerFactory.getLogger(ProjectStatsController.class);
+	private static final String SUBMITTED = "SUBMITTED";
+	private static final String APPROVED = "APPROVED";
+	private static final String REJECTED = "REJECTED";
```
Use INetworkStatus instead.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+			@PathVariable("projectUnixName") String projectUnixName, Model model, Principal principal)
+					throws JAXBException, QuadrigaStorageException, JSONException {
+
+		IProject project = projectManager.getProjectDetailsByUnixName(projectUnixName);
```
Take a look at this and try using the new annotation (you might have to merge devleop):
https://diging.atlassian.net/wiki/display/QUAD/Injecting+projects+by+unix+names

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+
+		if (project == null) {
+			return "auth/accessissue";
+		}
```
You can then also remove this if block.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+		List<INetwork> networks = networkmanager.getNetworksInProject(projectId);
+		List<IConceptStats> conceptsWithCount = null;
+		JSONArray submittedNetworkCount = null, approvedNetworkCount = null, rejectedNetworkCount  = null, workspaceCount = null;
+		String submittedNetworks = null, approvedNetworks = null, rejectedNetworks = null;
```
Read the style guide!

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+			
+			submittedNetworks = submittedNetworkCount.toString();
+			approvedNetworks = approvedNetworkCount.toString();
+			rejectedNetworks = rejectedNetworkCount.toString();
```
Instead of having another set of variables, just call 'toString' in when you add the counts to the model.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+
+		for(INetwork network : networks)
+		{
+			if(network.getStatus().equals(status) || status.equals("SUBMITTED"))
```
Why do you have this second condition here (`status.equals("SUBMITTED")`)?

bharathsharma replied:
There is no actual status called "SUBMITTED" . Basically its approved+rejected so if the request is for submitted, It will allow everything.

jdamerow replied:
ok, that makes sense. please put that as a comment into the code and in the javadoc.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+	 * @throws JSONException 
+	 *
+	 */
+	private JSONArray getContributionCountByStatus (List<INetwork> networks, String status) throws JSONException
```
This is logic that should be in your manager class.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -34,92 +54,162 @@
+		{
+			if(network.getStatus().equals(status) || status.equals("SUBMITTED"))
+			{
+				if(contributionCount.containsKey(sdf.format(network.getCreatedDate())))
```
You should assign this `sdf.format(network.getCreatedDate())` once into a variable, and then use the variable instead of repeating it over and over again.

diging-jenkins commented:
Can one of the admins verify this patch?

bharathsharma commented:
Addressed all issues .

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,91 @@
+        }
+        return workspaceCount;          
+    }
+
```
read the style guide!

bharathsharma replied:
fixed issue

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,78 @@
+                }
+            }
+        }
+        for(Entry<String, Integer> entry : contributionCount.entrySet()) {
```
sorry, I think I wasn't clear on that. Your controller decides what format it needs, your manager does the logic. So you controller should use a helper or helper method to create the json, you manager does the business logic.

bharathsharma replied:
got it..changed accordingly. Thanks

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,78 @@
+                contributionCount.put(workspaceCreationDate, 1);
+            }   
+        }
+        for(Entry<String, Integer> entry : contributionCount.entrySet()) {
```
same here.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,78 @@
+public class ContributionStatsManager implements IContributionStatsManager {
+
+    @Override
+    public JSONArray getContributionCountByStatus (List<INetwork> networks, String status) throws JSONException {
```
your method need javadoc

bharathsharma commented:
All issues addressed .

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -85,23 +130,26 @@ private JSONArray getTopConceptsJson(List<IConceptStats> conceptsList,
+        JSONArray submittedNetworkCount = null;;
+        JSONArray approvedNetworkCount = null;
+        JSONArray rejectedNetworkCount  = null;
+        JSONArray workspaceCount = null;
```
declare them where they are needed

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -122,8 +172,18 @@ public String showProjectStatistics(
+        model.addAttribute("submittedNetworksData",submittedNetworkCount.toString());
+        model.addAttribute("approvedNetworksData",approvedNetworkCount.toString());
+        model.addAttribute("rejectedNetworksData",rejectedNetworkCount.toString());
+        model.addAttribute("workspaceData",workspaceCount.toString());
```
and just move this bock up into if block

jdamerow replied:
and you would get a null pointer if there are no networks

bharathsharma replied:
checked for null if no networks .Thanks

bharathsharma commented:
All comments addressed as suggested. Thanks

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     */
+    @Override
+    public HashMap<String, Integer> getContributionCountByStatus (List<INetwork> networks, String status) throws JSONException {
```
remove jsonexception

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -72,6 +87,36 @@ private JSONArray getTopConceptsJson(List<IConceptStats> conceptsList,
+        JSONArray contributionsJson = new JSONArray();
+        HashMap<String, Integer> contributionCount = new HashMap<String, Integer>();
+        
+        contributionCount = contributionManager.getContributionCountByStatus(networks, status);
```
merge with line 93

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -72,6 +87,36 @@ private JSONArray getTopConceptsJson(List<IConceptStats> conceptsList,
+        JSONArray workspaceCountArray = new JSONArray();
+        HashMap<String, Integer> workspaceCount = new HashMap<String, Integer>();
+        
+        workspaceCount = contributionManager.getWorkspaceContribution(project);
```
merge with line 108

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/domain/impl/ContributionStatsManager.java:
```diff
@@ -0,0 +1,75 @@
+     *
+     */
+    @Override
+    public HashMap<String, Integer> getWorkspaceContribution (IProject project) throws JSONException {
```
remove exception

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -110,8 +153,10 @@ public String showProjectStatistics(
                 int cnt = getCount(conceptsWithCount);
                 jArray = getTopConceptsJson(conceptsWithCount.subList(0, cnt),
```
jArray should be declared here, not in line 152.

jdamerow commented on Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java:
```diff
@@ -110,8 +153,10 @@ public String showProjectStatistics(
+
                 model.addAttribute("networkid", "\"\"");
 
             } catch (JSONException e) {
```
the exceptions are all over the place. some you catch like here, some you just let your controller throw. Just catch all of them and show an error message as already done below.

jdamerow commented:
ok to test

jdamerow commented:
ok to test

jdamerow commented:
Let's don't have floats on the y axis. Doesn't make much sense, since you can't submit half a network:
![image](https://cloud.githubusercontent.com/assets/8881141/14830234/7e4789b4-0ba4-11e6-96d7-c1dab5694a3e.png)

Also, if there are no datapoints, just show a message saying that there is nothing to show. Not an empty diagram.

And why is there just nothing for rejected networks and workspaces?
![image](https://cloud.githubusercontent.com/assets/8881141/14830251/a43ecfb0-0ba4-11e6-94d8-e778e5bc7e4f.png)

jdamerow commented:
Also, the hover text for concepts is cut off again:
![image](https://cloud.githubusercontent.com/assets/8881141/14830278/c80a9f5a-0ba4-11e6-8186-2b32e5c15560.png)

bharathsharma commented:
1.Fixed diagram position and display issues. 
2.Currently if there is a single point,  say 4 networks are submitted on same date 26-April-2016, I am just displaying that point on graph even though there wont be much of pattern to show.
3.Addressed all remaining issues.

Thanks.

diging-jenkins commented:
Can one of the admins verify this patch?

jdamerow commented:
ok to test

jdamerow commented:
Hm, but what happens here:
![image](https://cloud.githubusercontent.com/assets/8881141/14867784/12b7d178-0c7f-11e6-976e-20b94e72bcdd.png)
![image](https://cloud.githubusercontent.com/assets/8881141/14867807/2a84bf3c-0c7f-11e6-93a5-098a336cf90d.png)

Also, the first time the page is loaded (when the cache hasn't kicked in yet), I don't get any diagrams and just headings.

bharathsharma commented:
Thats because we have only one record in it .If you move the mouse pointer near date, It will show the count. Ill see how I can make it more appealing. 

Ill also have a look at the cache issue.

Thanks.

jdamerow commented:
ok, I'll merge it for now. If you make more changes, please make another pull request.
